Code review

KatarzynaMorys opened the pull request.

KatarzynaMorys committed:
Specifying a Java version

AnnaMorys committed:
Adding home page

AnnaMorys committed:
Adding template pages with header

KatarzynaMorys committed:
Modifying header and adding footer

KatarzynaMorys committed:
Adding pages templates with header and footer

AnnaMorys committed:
Climbing gear: adding link buttons

KatarzynaMorys committed:
Modifying womenClothing.jsp and climbingGear.jsp

AnnaMorys committed:
Refactoring style, adding css files to resources

KatarzynaMorys committed:
Modifying womenClothing.jsp and menClothing.jsp

KatarzynaMorys committed:
Connecting to postgresql database

AnnaMorys committed:
Creating database tables, adding entity and repositories.
Adding templates for climbing gear web page
Refactoring controller

AnnaMorys committed:
Creating webpage for admin to add products

KatarzynaMorys committed:
Adding entities: Customer, Address, Order. Create database tables

KatarzynaMorys committed:
Merge remote-tracking branch 'origin/master'

KatarzynaMorys committed:
correct entities id names

KatarzynaMorys committed:
implement customer register

KatarzynaMorys committed:
adding bootstrap stylesheet link, scripts, jstl, login, register, account page

AnnaMorys committed:
add unique constraint to category and brand name

AnnaMorys committed:
refactoring productList.jsp

AnnaMorys committed:
creating adminHome.jsp to add products to database

AnnaMorys committed:
Merge branch 'admin_webpage'

AnnaMorys committed:
moving to new packages

AnnaMorys committed:
Bug fix -> wrong category and brand adding to database,
refactoring style adminHome.jsp

AnnaMorys committed:
Merge pull request #1 from KatarzynaMorys/admin_webpage

Bug fix -> wrong category and brand adding to database,

KatarzynaMorys committed:
Added jackson-module-jaxb-annotations dependency

KatarzynaMorys committed:
Added MANIFEST.MF

KatarzynaMorys committed:
delete unnecessary file

AnnaMorys committed:
changes in database:
adding column description and image to product;
creating table orders_products

AnnaMorys committed:
Adding new fields do product entity
Editing adminHome.jsp and AdminController

AnnaMorys committed:
Filter products by category

AnnaMorys committed:
Merge pull request #2 from KatarzynaMorys/admin_webpage

Admin webpage

AnnaMorys committed:
refactoring jsp files

AnnaMorys committed:
- extending CategoryRepository and ProductRepository by
findCategoryByCategoryName and findProductsByCategory_CategoryName
- refactoring CategoryRepository

AnnaMorys committed:
Merge pull request #3 from KatarzynaMorys/category

Category

KatarzynaMorys committed:
Adding table roles, employees and relations

KatarzynaMorys committed:
Configure role-based access control: USER, ADMIN

KatarzynaMorys committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/resources/db/changelog/changelog-master.xml

AnnaMorys committed:
adding men, women controller and jsp files

AnnaMorys committed:
adding shopping button to category.jsp, renaming hats&accessories.jsp.jsp

AnnaMorys committed:
Merge pull request #4 from KatarzynaMorys/clothing

Clothing

KatarzynaMorys committed:
Create findRoleByNameIsLike(String name) method & refactor CustomerRegistrationController

AnnaMorys committed:
editing database, adding product_subtype column, changes in the related files

AnnaMorys committed:
filter products

AnnaMorys committed:
Merge pull request #5 from KatarzynaMorys/view_products

View products

AnnaMorys committed:
fix bug -> wrong product type save in the database

KatarzynaMorys committed:
Implement shopping bag

KatarzynaMorys committed:
Correcting jsp files

KatarzynaMorys committed:
Merge branch 'shoppingBag'

# Conflicts:
#	src/main/webapp/WEB-INF/jsp/category.jsp

KatarzynaMorys committed:
Refactoring jsp files, adding thanks.jsp

KatarzynaMorys committed:
OrdersProducts entity + repository

KatarzynaMorys committed:
Removing timestamp

KatarzynaMorys committed:
Implement adding contact details

KatarzynaMorys committed:
Implement order

KatarzynaMorys committed:
Merge pull request #7 from KatarzynaMorys/order

Order

AnnaMorys committed:
refactor footer: removing comment

AnnaMorys committed:
changing in database: adding customer to reviews table

AnnaMorys committed:
adding review functionality

AnnaMorys committed:
Merge branch 'master' into review

AnnaMorys committed:
fix bug -> wrong redirect

AnnaMorys committed:
Rating value displayed

AnnaMorys committed:
Merge pull request #8 from KatarzynaMorys/rating

Rating value displayed

karola808 commented on src/main/java/pl/sda/climbing_shop/configuration/SecurityConfiguration.java:
```diff
@@ -0,0 +1,38 @@
+    protected void configure(HttpSecurity http) throws Exception {
+
+        http
+                .csrf().disable()
```
czemu csrf jest wylaczony?

karola808 commented on src/main/java/pl/sda/climbing_shop/configuration/UserDetailsServiceImplForCustomers.java:
```diff
@@ -0,0 +1,79 @@
+                        .collect(Collectors.toList()))
+                .username(customer.getEmail())
+                .password(customer.getHashPassword())
+                .accountNonExpired(true)
+                .accountNonLocked(true)
+                .credentialsNonExpired(true)
+                .enabled(true)
```
czy te rzeczy nie powinny być ciągniete z bazy? czy moze to nie przewidzane w pierwszej fazie  ? ;P

karola808 replied:
dobra, ogarnelam, że spirngowy inteface userdetails wymaga ;)

karola808 commented on src/main/java/pl/sda/climbing_shop/configuration/UserDetailsServiceImplForEmployees.java:
```diff
@@ -0,0 +1,70 @@
+    private final EmployeeRepository employeeRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username) throws UsernameNotFoundException {
```
może warto by ugenerycznić tą metode loadUserByUsername<T>  albo klase UserDetailsService <T>   (nie pamietam jak w javie to jest)
gdzie T moze byc employyeRepository lub customerRepository

karola808 commented on src/main/java/pl/sda/climbing_shop/controller/WomenClothingController.java:
```diff
@@ -0,0 +1,65 @@
+    }
+
+    @PostMapping("/{categoryName}")
+    public String filterWomenClothingCategory(@PathVariable("categoryName") String categoryName,
```
post sugeruje zmiane satnu aplikacji. Filtrowanie nie powinno zmieniać stanu/danych. Wiec albo get / albo zmiana nazwy metody ;)

karola808 commented on src/main/java/pl/sda/climbing_shop/controller/WomenClothingController.java:
```diff
@@ -0,0 +1,65 @@
+    public String filterWomenClothingCategory(@PathVariable("categoryName") String categoryName,
+                                              @ModelAttribute("product") Product product) {
+
+        return "redirect:/womenClothing/" + filterProducts(categoryName, product);
```
ten post chyba powinine byc getem, jako ze nie zmienia stanu aplikacji. Swoja droga, czemu te filtrowanie jest zorbione na zasadzie redirecta to zbudowanego urla?

karola808 commented on src/main/java/pl/sda/climbing_shop/controller/ShoppingBagController.java:
```diff
@@ -0,0 +1,160 @@
+            return storedQuantity;
+        else {
+            return orderedQuantity;
+        }
```
syntax sugar:  return orderedQuantity > storedQuantity?  storedQuantity : orderedQuantity

karola808 commented on src/main/java/pl/sda/climbing_shop/controller/ClimbingGearController.java:
```diff
@@ -0,0 +1,134 @@
+        Authentication authentication = SecurityContextHolder.getContext().getAuthentication();
+
+        if (authentication.getName().equals("anonymousUser")) {
+            return "redirect:/login";
```
tak z ciekawości, bo nie wiem jak działa spirngowa authentykacja. Czy powyzsze nie sprawi, że nie bedzie mogl istniec user o nazwie "anonymousUser" ?

karola808 commented on src/main/java/pl/sda/climbing_shop/controller/ClimbingGearController.java:
```diff
@@ -0,0 +1,134 @@
+                                           @RequestParam(required = false) String color,
+                                           @RequestParam(required = false) String brand) {
+
+        return viewProducts(categoryName, model, type, size, color, brand, this.categoryRepository, this.productRepository);
```
moze warto porworzyc jakies Viewmodele  na takie metody?
Mam na mysli jakas klase ktora to wszystko bedzie trzymac

karola808 commented on src/main/java/pl/sda/climbing_shop/controller/AdminController.java:
```diff
@@ -0,0 +1,112 @@
+                .discountValue(formData.getDiscountValue())
+                .build();
+
+        Category category = Category.builder()
```
jest sens uzywania builderów do rpzypisania jednej wartyosci ?

karola808 commented on src/main/java/pl/sda/climbing_shop/controller/AdminController.java:
```diff
@@ -0,0 +1,112 @@
+            newProduct.setBrand(brand);
+        }
+
+        this.priceRepository.save(price);
```
jak w tej metodzie jest zachowana transakcyjnosc? W sensie jak sie powiedzie zapis do priceRepository a nie powiedzie zapis w producktRepository to jaki bedzie wynik? idealnie zeby nic sie nie zapisalo, jesli czesciowo zapisane, a czesciowio nie, to troche słabo

karola808 commented on src/main/java/pl/sda/climbing_shop/controller/AdminController.java:
```diff
@@ -0,0 +1,112 @@
+            if (tempC.getCategoryName().equals(category.getCategoryName())) {
+                performSave = false;
+                product.setCategory(tempC);
+                break;
```
metoda has... nie powinna zmieniać nic a tylko sprawdzać. Zastanawiam się czy tego checka nie można wyrzucić bezpośrednio do metody która woła to metode i zastpiać  wyrażeniem stream

AnnaMorys committed:
ClimbingShopApplication

KatarzynaMorys committed:
Add @NoArgsConstructor and @AllArgsConstructor in ProductFormData

AnnaMorys committed:
- create View - helper for ClimbingGearController, WomenClothingController and MenClothingController;
- ahref correction in menClothing.jsp;

AnnaMorys committed:
Changes in jsp files:
- including template webpages for all products: clothing.jsp or gear.jsp;
- creating new filter for clothing

AnnaMorys committed:
- adding new query for clothing filtering
- creating View - helper for ClimbingGearController and ClothingController

KatarzynaMorys committed:
Change css files to display products

KatarzynaMorys committed:
Merge pull request #10 from KatarzynaMorys/view-products

Change css files to display products

AnnaMorys committed:
Creating a new product filtering.
TODO: some problems occur during brand selection

AnnaMorys committed:
formatting jsp form
refactoring view of jsp files
renaming file filter.jsp to gearFilter.jsp

AnnaMorys committed:
filtering products by selecting parameters

AnnaMorys committed:
Merge pull request #11 from KatarzynaMorys/filtering

Filtering

AnnaMorys committed:
bootstrap responsive on mobiles

AnnaMorys committed:
bootstrap responsive on mobiles, menu display on mobiles

AnnaMorys committed:
-> new men image display
-> changing method filtering to GET

AnnaMorys committed:
-> new way of products filtering, saving filtering parameters in session attribute "filterMap"
-> deleting unused searching methods from ProductRepository
-> new method getValue() in Product class, returning field value by name
-> @GetMapping used in clothing and gear controllers
-> TODO: removing session attribute "filterMap" during changing URL by customer

AnnaMorys committed:
-> removing session attribute "filterMap" during changing URL by customer

AnnaMorys committed:
Merge pull request #12 from KatarzynaMorys/productsfiltering

Products filtering

KatarzynaMorys committed:
Modiyfing customer account page

KatarzynaMorys committed:
Adding methods to change name, email, password, contact details

KatarzynaMorys committed:
Merge branch 'customer_account'

KatarzynaMorys committed:
Changes in jsp files, making website responsive to mobiles

KatarzynaMorys committed:
Merge pull request #13 from KatarzynaMorys/view

Changes in jsp files, making website responsive to mobiles

KatarzynaMorys committed:
Implement AuthenticationSuccessHandler and AuthenticationFailureHandler

KatarzynaMorys committed:
Merge pull request #14 from KatarzynaMorys/authentication_handler

Implement AuthenticationSuccessHandler and AuthenticationFailureHandler
